### PR TITLE
dds ls --json

### PR DESCRIPTION
**Pull request by alneberg**
Sorry for a very messy PR. Most of the changes are moving around the methods inside the data lister to adhere to the standard we have in most other classes.

### CODE REVIEW

File: dds_cli/data_lister.py [hunk 1]
```diff
@@ -338,12 +214,7 @@ class FileTree:
             subtrees: List[Union["FileTree", Tuple[str, str]]] = None
             name: str = None
 
-        def _construct_file_tree(folder: str, basename: str) -> Tuple[FileTree, int, int]:
-            """
-            Recurses through the project directories.
-
-            Constructs a file tree by subsequent calls to the API
-            """
+        def __api_call_list_files(folder):
```
Comment (inaod568) [inline dds_cli/data_lister.py:217]:
```
(dds_venv) $ dds ls project_1 --tree
     ︵ 
 ︵ (  )   ︵ 
(  ) ) (  (  )   SciLifeLab Data Delivery System 
 ︶  (  ) ) (    https://www.scilifelab.se/data 
      ︶ (  )    Version 0.1.dev0 
          ︶

Traceback (most recent call last):
  File "/Users/inaod568/repos/DDS/dds_venv/bin/dds", line 33, in <module>
    sys.exit(load_entry_point('dds-cli', 'console_scripts', 'dds')())
  File "/Users/inaod568/repos/DDS/dds_venv/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/inaod568/repos/DDS/dds_venv/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/inaod568/repos/DDS/dds_venv/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/inaod568/repos/DDS/dds_venv/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/inaod568/repos/DDS/dds_venv/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/inaod568/repos/DDS/dds_venv/lib/python3.8/site-packages/click/decorators.py", line 33, in new_func
    return f(get_current_context().obj, *args, **kwargs)
  File "/Users/inaod568/repos/DDS/dds_cli/dds_cli/__main__.py", line 377, in ls
    folders = lister.list_recursive(show_size=size)
  File "/Users/inaod568/repos/DDS/dds_cli/dds_cli/data_lister.py", line 342, in list_recursive
    file_tree, max_string, max_size = __construct_file_tree(
  File "/Users/inaod568/repos/DDS/dds_cli/dds_cli/data_lister.py", line 245, in __construct_file_tree
    sorted_files_folders = __api_call_list_files(folder)
  File "/Users/inaod568/repos/DDS/dds_cli/dds_cli/data_lister.py", line 231, in __api_call_list_files
    sorted_files_folders = sorted(resp_json["files_folders"], key=lambda f: f["name"])
KeyError: 'files_folders'
```

Comment (inaod568) [reply dds_cli/data_lister.py:217]: There needs to be a check for if the information is returned (if `files_folders` is empty)

**Review by inaod568 — CHANGES_REQUESTED** [review]
Looks good! 
Have one comment atm. 

Also, the file sizes are listed in a bit of a strange way but that's another card probably. Try uploading a folder with both small and larger files and then doing `dds ls --tree --size`

File: dds_cli/data_lister.py (hunk 1 above)
Comment (alneberg) [reply dds_cli/data_lister.py:217]: Ok, should be fixed now!

**Review by inaod568 — APPROVED** [review]
Looks good!

**Pull request merged**